File: packages/python/dirsql/cli/interpret/write_message.py
"""Single-line NDJSON writer for the `dirsql interpret` helper."""

from __future__ import annotations

import json
import sys
from typing import Any


def write_message(msg: dict[str, Any]) -> None:
    """Write one NDJSON line to stdout and flush.

    The flush is required so the orchestrator on the other side of the
    pipe sees the message immediately; without it the buffered stream
    would deliver lines in batches.
    """
    sys.stdout.write(json.dumps(msg) + "\n")
    sys.stdout.flush()

File: packages/ts/src/cli/interpret/writeMessage.ts
// Single-line NDJSON writer. Each protocol message is one JSON object
// followed by `\n`; `process.stdout.write` flushes to the underlying
// pipe synchronously on most platforms when the stream is a pipe (as
// it is when the orchestrator spawns this process).

export function writeMessage(msg: unknown): void {
  process.stdout.write(`${JSON.stringify(msg)}\n`);
}
